rfcs/042-editable-programmatic-descriptions.md

```diff
@@ -0,0 +1,100 @@
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000)
+
+# Editable Programmatic Descriptions
+
+## Summary
+
+Programmatic descriptions are a useful way for Amundsen users to store and display business specific metadata about their tables in Amundsen. This feature allows for some customization of Amundsen without customizing the underlying software. Currently, this feature focuses on the machine generated metadata use case. This type of data can be extracted from the source system of truth and ingested into Amundsen via databuilder.
+
+The same need exists for business specific human generated metadata. This type of data does not exist in a machine generated form that can be easily extracted from a source system. Instead, users need to add/edit this metadata directly in the Amundsen UI. 
+
+## Motivation
+
+This feature allows users to configure their Amundsen installation to support one or more editable programmatic descriptions. These editable programmatic descriptions allow users to enrich the metadata displayed on the table details page with additional business specific human generated information. This feature will allow them to add/edit these programmatic descriptions directly in the UI.
+
+## Guide-level Explanation (aka Product Details)
+
+An editable programmatic description is the same as a regular programmatic description, except that it is maintained directly in the UI by a human instead of by a scheduled databuilder job. For example, users could configure an editable programmatic description with a source of data_usage_policy and then update the data_usage_policy on individual table detail pages with a link to the applicable data usage policy on their company's content management site.
+
+Similar to table descriptions, if a table is disabled for editing, then so are the editable programmatic descriptions.
+
+Databuilder jobs may still be useful on a one-off (i.e. non-scheduled basis) to seed Amundsen with manually curated data or perform one-off data conversations.
+
+There will be no impact to existing programmatic descriptions. They will remain non-editable via the UI unless users intentionally update the existing configuration in their Amundsen installation to make them editable.
+
+## UI/UX-level Explanation
+
+Editable programmatic descriptions would be displayed alongside programmatic descriptions in the same way they are displayed today. They will be rendered as markdown, located in one of three sections (left, right, other) in the page and sorted according to display_order.
+
+There are two differences between editable and non-editable programmatic descriptions:
+1. They are editable (if the table is editable)
+2. They will always be displayed. If there is not a value then placeholder text will be displayed instead.
+
+_**Note:** The pink text is only visible in the mocks to call out design details and changes._
```

Comment:
this is a question for the UI proposal - how will existing Amundsen users (not the team that is deploying Amundsen) `know` that given programmatic description is editable if `edit` icon just appears on hover?

Reply:
That is a good point. We should always display the edit icon. This is the default behavior of `EditableSection` today. I will update the mock.

Reply:
Updated